src/Symfony/Component/Security/Http/Authenticator/Passport/Badge/UserBadge.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\Passport\Badge;
+
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Http\EventListener\UserProviderListener;
+
+class UserBadge implements BadgeInterface
+{
+    private $username;
+    private $userLoader;
+    private $user;
+
+    public function __construct(?string $username, ?callable $userLoader = null)
+    {
+        $this->username = $username;
+        $this->userLoader = $userLoader;
+    }
+
+    public function getUsername(): string
```

Comment:
As it's written now, this could be nullable. But do we want to allow the first arg to the constructor to be nullable?